src/transforms/repair.jl

```diff
@@ -20,6 +20,75 @@ Perform repairing operation with code `K`.
 """
 struct Repair{K} <: StatelessGeometricTransform end
 
+
+# --------------
+# OPERATION (0)
+# --------------
+
+function apply(::Repair{0}, mesh)
+  points  = vertices(mesh)
+  npoints = nvertices(mesh)
+
+  # store indicators of duplicated vertices
+  dups = fill(false, npoints)
+
+  # dictionary mapping old to new indices
+  inds = Dict{Int,Int}()
+
+  count = 0
+
+  for v in 1:(npoints-1)
+    if !dups[v]
+      # current point
+      pt = points[v]
+      # points after `pt`
+      tail = points[(v+1):npoints]
+      # indices of points equal to `pt` in `tail`
+      duplicates = v .+ findall(==(pt), tail)
+      dups[duplicates] .= true
+      push!(duplicates, v)
+      count += 1
+      for i in duplicates
+        inds[i] = count
+      end
+    end
+  end
+  
+  # if last vertex is not a duplicate, add it to the dictionary
+  if npoints ∉ keys(inds) 
+    inds[npoints] = count
+  end
+
+  # get the elements (faces)
+  topo = topology(mesh)
+  ∂₂₀ = Boundary{2,0}(topo)
+  nelems = nelements(topo)
```

Comment:
since we are only using Boundary{2,0}, we should avoid writing the code in terms of relations, and instead just loop over elements(topo) which has the same result.

Reply:
I don't understand what you want to mean. Could you elaborate? What should I change?

Reply:
You can do `for e in elements(topology(mesh))` to get the `Connectivity` objects and then use `indices(e)` to get the indices returned by the Boundary{2,0}. No need to use boundary relation in the implementation when it just uses Boundary{2,0}.